slm_porosity/heat_transfer.f90

```diff
@@ -826,4 +906,25 @@ FUNCTION Spline_cubic (r_p, l_p, fr_p, fl_p, dfr_p, dfl_p)
       - fr_p*l_p**3 - r_p**2*(3*fl_p*l_p - dfl_p*l_p**2 + dfr_p*l_p**2))/(r_p-l_p)**3
   END FUNCTION Spline_cubic
 
+  FUNCTION three_parameter_connection (temperature, liquid_fraction, der_solid, der_liquid, fusion, is_D)
+    IMPLICIT NONE
+    REAL (pr), INTENT(IN) ::  temperature(:), liquid_fraction(:)
+    REAL (pr) :: three_parameter_connection(SIZE(temperature))
+    REAL (pr) :: der_solid, der_liquid, fusion
+    LOGICAL, OPTIONAL, INTENT(IN) :: is_D   !derivative over phi for DRHS and DRHS_diag
+
+    IF (.NOT.PRESENT(is_D).OR.(.NOT.is_D)) THEN
+      three_parameter_connection = 1.0_pr + der_solid*temperature + &
+        (fusion + (der_liquid - der_solid)*(temperature - 1.0_pr))*liquid_fraction
+    ELSE
+      three_parameter_connection = (fusion + (der_liquid - der_solid)*(temperature - 1.0_pr))
```

Comment:
What is this? It should be `df/dh`, is not it?

Reply:
In this case, we should use `Dtemperature` and `Dliquid_fraction`. To avoid confusion, it can be a separate function `Dthree_parameter_model(Dtemperature, Dliquid_fraction, Dsolid, Dliquid, fusion_jump)` or  `three_parameter_model(enthalpy, Dsolid, Dliquid, fusion_jump, is_D)`.